docs/faq/repositories/how-do-i-block-merging-prs-using-codacy-as-a-quality-gate.md

```diff
@@ -0,0 +1,20 @@
+---
+description: Configure Codacy to block merging pull requests that don't meet your quality standards.
+---
+
+# How do I block merging pull requests using Codacy as a quality gate?
+
+You can block merging pull requests until they pass the Codacy quality gate. This ensures the quality of the changes to your codebase, such as to prevent introducing security issues or adding code that isn't covered by tests.
+
+To configure Codacy to block merging pull requests that don't meet your quality standards:
+
+1.  Set up your repository so that:
+
+    -   **Codacy** reports status checks for pull requests on your Git provider
+    -   **Your Git provider** blocks merging pull requests if they don't pass the Codacy status check
+
+    To do this, follow the instructions that apply to [GitHub](../../repositories-configure/integrations/github-integration.md#configuring-the-github-integration), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#configuring-the-gitlab-integration), or [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#configuring-the-bitbucket-integration), depending on your Git provider.
+
+1.  Optionally, if you want to block merging pull requests that don't meet your code coverage standards, make sure that you [set up coverage on your repository](../../coverage-reporter/index.md).
```

Comment:
Something that might be worth clarifying is that users need to enable the coverage Quality Setting option. Otherwise, they won't receive the resulting coverage checks in the git provider - which, unlike the check for resulting from the code quality analysis, is not enabled by default.

I don't think that that step is explicit here or in the linked coverage guide.

We are mentioning it in the specific integration pages but might be easily missed:
> To get the coverage status you must [set up coverage on your repository](../../coverage-reporter/index.md) and **enable the coverage rule on the quality settings**.

Reply:
I second this, it's a question that comes up quite often in Support

Reply:
See also the extra recommendation mentioned by @jamescodacy on this [Slack thread](https://codacy.slack.com/archives/CA5GWEXSN/p1643299700029900?thread_ts=1643205653.029600&cid=CA5GWEXSN), and that I'm also going to apply. Great feedback! :raised_hands:

Reply:
After thinking some more about the feedback I received from you and from @jamescodacy, I arrived at the conclusion that the process to set up this use case is a bit more involved and it would help users if we:

- Provided an overview of the end-to-end process
- Included some extra details and advice on relevant steps
- Tried to ensure that users would see the full instructions on how to set up the use case before using Codacy to block pull requests

Because of this, I've reworked the FAQ page to better guide users and pointed other pages across the documentation to this centralized location to help ensure that users get to see the full instructions.

If you have the time please have another look! :pray: 

https://deploy-preview-995--docs-codacy.netlify.app/faq/repositories/how-do-i-block-merging-prs-using-codacy-as-a-quality-gate/